source/val/BasicBlock.h

```diff
@@ -31,6 +31,7 @@
 
 #include <cstdint>
 
+#include <algorithm>
```

Comment:
I like to add non-trivial functions in the `cpp` files to avoid bringing in large headers. hmm, it seems that `algorithm` is being included in `validate.h` so this will not really hurt compile times. Your call.